### ISSUE TITLE
Cross-account functionality - deploying same set of policies across a group of accounts

### ISSUE THREAD

**Issue by caversa** [cloud-custodian/cloud-custodian#2124]
Before I saw the c7n-org functionality (I didn't see that in the online doc but saw it in github), I wrote a wrapper myself that provides cross-account functionality.  The structure is quite similar to what you designed, but it looks a little different in the fact that I desired to provide account groupings, and define which policies were related to that group.  We've got Development/Test accounts versus Production accounts, and  we terminate non-compliant resources in the Dev/Test, and only notify in the Production accounts.   I thought I'd share this in case this type of thing could be helpful to the project, or if others had a similar need to mine.  I use the same cross-account role for many accounts, so the full arn is determined based upon the target account.  One or more policies can be specified that are related to a group.  I create a new account 'group' based upon the combo of policies and the assume role.  An account can belong to more than one group if necessary.  This helps me manage the policy inventory a bit better.  If I have a new policy, I can add it to a group and pass in the policy file name.  If I have a new account, I can add it to the group and pass in the account parameter.  Or, I can deploy all specified policies to all the group's accounts/regions.  I also added a 'plan' parameter to print out the custodian commands without actually calling custodian.

[custodian_wrapper.txt](https://github.com/capitalone/cloud-custodian/files/1825337/custodian_wrapper.txt)

Below is my sample yaml file:

```
# Only requires account number. Alias is for informational only (prints out) when deploying to the account
# When no region is specified, a policy will run again us-east-1.

groups:
- name: devops-devtest
  assume-role: custodian-deploy-assume-role
  policies:
      - tagging_compliance_terminate
      - unassociated_eip
      - unattached_volumes
      - copy_tags_to_ebs
      - delete_expired_resources
  accounts:
      - account: 123456789012
      - account: 234567890123

      - account: 345678901234
        alias: test-10000
        regions:
          - us-east-1
          - us-east-2
          - us-west-2
          - ap-southeast-2
          - eu-west-1
          - eu-west-2
          - ca-central-1
```

**Comment by kapilt** [comment]
Thanks for your sharing your script. A couple of items

 - c7n-org does support running against a subset of accounts within a given accounts file by subset.
 - atm for c7n-org, a subset of policies for different environments is typically done via tag filter to accounts and a separate set of policy files for each set of accounts.
 - interpreting cross account role seems like a reasonable extension/default perhaps just given role name in the account and constructing the rest of the Arn, or using the default applied. we currently support several more modes due to various real world variance and usage modes (profiles to support mfa auth or human users, etc).
 - policies also support tags, so there might be some value in exploring that as a separate filter on policy execution.
 - c7n-org could use some more docs and highlighting in the main docs, it also has support for executing arbitrary scripts/programs with sdk credentials in a given account/region.

**Comment by caversa** [comment]
Thanks very much for the feedback, Kapil.  For the second bullet point, I think I will continue to use the custom wrapper I created (at least for now), as I'm organizing the yml files into chunks of functionality, and I can reuse that same yml file for multiple account groupings and not duplicate the policy yml itself.   Since I'm not the only one going to be deploying Custodian policies (as we've got more than one team responsible for accounts) I like organizing it in a way that others can selectively consume a policy 'grouping' I've written (that I document), depending on whether they desire that functionality (without them needing to be Custodian experts themselves).  

That and the cross-account role determination as well (but that is probably low-hanging fruit for adding).  I'll definitely monitor what you guys do with the cross-account functionality, as I would rather use your baseline rather than a custom thing.    But this was a main hurdle preventing me from rolling Custodian out at a bigger scale for our needs - and I'm a huge fan of it.  Thanks for all the work you do with it!

Carrie
